src/main/java/mods/railcraft/common/fluids/FluidTools.java

```diff
@@ -198,90 +210,108 @@ public static ProcessState processContainer(IInventory inv, StandardTank tank, b
-//            FluidStack drain = fluidHandler.drain(null, fill.amount, false);
-//            if (drain != null && drain.amount == fill.amount) {
-//                fill = FluidItemHelper.fillContainer(input, drain);
-//                if (fill.container != null && fill.amount == drain.amount) {
-//                    fluidHandler.drain(null, fill.amount, true);
-//                    storeContainer(inv, inputSlot, outputSlot, fill.container);
-//                }
-//                return true;
-//            }
-//        }
-//        return false;
-//    }
-//
-//    public static boolean drainContainers(IFluidHandler fluidHandler, IInventory inv, int inputSlot, int outputSlot) {
-//        ItemStack input = inv.getStackInSlot(inputSlot);
-//        ItemStack output = inv.getStackInSlot(outputSlot);
-//        if (input != null) {
-//            FluidItemHelper.DrainReturn drain = FluidItemHelper.drainContainer(input, PROCESS_VOLUME);
-//            if (drain.fluidDrained != null && (drain.container == null || hasPlaceToPutContainer(output, drain.container))) {
-//                int used = fluidHandler.fill(null, drain.fluidDrained, false);
-//                if (drain.isAtomic ? used == drain.fluidDrained.amount : drain.fluidDrained.amount > 0) {
-//                    fluidHandler.fill(null, drain.fluidDrained, true);
-//                    storeContainer(inv, inputSlot, outputSlot, drain.container);
-//                    return true;
-//                }
-//            }
-//        }
-//        return false;
-//    }
-//
-//    private static boolean hasPlaceToPutContainer(@Nullable ItemStack output, @Nullable ItemStack container) {
-//        return output == null || output.stackSize < output.getMaxStackSize() && InvTools.isItemEqual(container, output);
-//    }
-//
-//    /**
-//     * We can assume that if null is passed for the container that the container
-//     * was consumed by the process and we should just remove the input container.
-//     */
-//    private static void storeContainer(IInventory inv, int inputSlot, int outputSlot, @Nullable ItemStack container) {
-//        if (container == null) {
-//            inv.decrStackSize(inputSlot, 1);
-//            return;
-//        }
-//        ItemStack output = inv.getStackInSlot(outputSlot);
-//        if (output == null)
-//            inv.setInventorySlotContents(outputSlot, container);
-//        else
-//            output.stackSize++;
-//        inv.decrStackSize(inputSlot, 1);
-//    }
+
+    /**
+     * Process containers in input/output slot like tha in the tank cart.
+     *
+     * @param tank       Fluid tank
+     * @param inv        The inventory that contains input/output slots
+     * @param inputSlot  The input slot number
+     * @param outputSlot The output slot number
+     * @return {@code true} if changes have been done to the tank
+     */
+    public static boolean processContainers(StandardTank tank, IInventory inv, int inputSlot, int outputSlot) {
```

Comment:
@CovertJaguar For that claim, we need to find one of those containers and test. Extrautil doesn't have one, we need to find others!